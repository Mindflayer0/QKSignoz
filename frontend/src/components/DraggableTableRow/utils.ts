import { DragSourceMonitor, DropTargetMonitor } from 'react-dnd';

export function dropHandler(
	monitor: DropTargetMonitor,
	index: number,
): { isOver: boolean } | Record<string, unknown> {
	const { index: draggedId }: never = monitor.getItem();

	if (draggedId === index) {
		return {};
	}
	return {
		isOver: monitor.isOver(),
	};
}

export function dragHandler(
	monitor: DragSourceMonitor,
): { isDragging: boolean } {
	return {
		isDragging: monitor.isDragging(),
	};
}
