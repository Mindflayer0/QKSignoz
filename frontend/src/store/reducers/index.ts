import { combineReducers } from 'redux';

import appReducer from './app';
import dashboardReducer from './dashboard';
import globalTimeReducer from './global';
import { LogsReducer } from './logs';
import metricsReducers from './metric';
import { PipelineReducer } from './pipeline';
import { ServiceMapReducer } from './serviceMap';
import traceReducer from './trace';
import { usageDataReducer } from './usage';

const reducers = combineReducers({
	traces: traceReducer,
	usageDate: usageDataReducer,
	globalTime: globalTimeReducer,
	serviceMap: ServiceMapReducer,
	dashboards: dashboardReducer,
	app: appReducer,
	metrics: metricsReducers,
	logs: LogsReducer,
	pipeline: PipelineReducer,
});

export type AppState = ReturnType<typeof reducers>;

export default reducers;
