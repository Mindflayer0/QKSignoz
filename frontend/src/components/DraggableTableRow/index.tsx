import React, { useRef } from 'react';
import { useDrag, useDrop } from 'react-dnd';
import { v4 } from 'uuid';

import { dragHandler, dropHandler } from './utils';

const type = 'DraggableTableRow';

function DraggableTableRow({
	index,
	moveRow,
	className,
	style,
	...restProps
}: DraggableTableRowProps): JSX.Element {
	const ref = useRef<HTMLTableRowElement>(null);
	const [, drop] = useDrop({
		accept: type,
		collect: (monitor) => dropHandler(monitor),
		drop: (item: { index: number }) => moveRow(item.index, index),
	});
	const [, drag] = useDrag({
		type,
		item: { index },
		collect: (monitor) => dragHandler(monitor),
	});
	drop(drag(ref));

	return (
		<tr
			key={v4()}
			ref={ref}
			className={className}
			style={{ ...style }}
			// eslint-disable-next-line react/jsx-props-no-spreading
			{...restProps}
		/>
	);
}

interface DraggableTableRowProps
	extends React.HTMLAttributes<HTMLTableRowElement> {
	index: number;
	moveRow: (dragIndex: number, hoverIndex: number) => void;
}

export default DraggableTableRow;
