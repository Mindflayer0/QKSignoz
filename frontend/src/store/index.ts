import { applyMiddleware, compose, createStore } from 'redux';
import type { ThunkMiddleware } from 'redux-thunk';
import thunk from 'redux-thunk';

import reducers, { AppState } from './reducers';

const composeEnhancers =
	(window && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) || compose;

const store = createStore(
	reducers,
	// @TODO Add Type for AppActions also
	composeEnhancers(applyMiddleware(thunk as ThunkMiddleware<AppState>)),
);

export default store;
