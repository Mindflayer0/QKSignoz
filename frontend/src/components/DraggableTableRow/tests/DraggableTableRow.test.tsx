import { render } from '@testing-library/react';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import i18n from 'ReactI18';
import store from 'store';

import DraggableTableRow from '..';

jest.mock('react-dnd', () => ({
	useDrop: jest.fn().mockImplementation(() => [jest.fn(), jest.fn(), jest.fn()]),
	useDrag: jest.fn().mockImplementation(() => [jest.fn(), jest.fn(), jest.fn()]),
}));

describe('DraggableTableRow Snapshot test', () => {
	it('should render DraggableTableRow', () => {
		const { asFragment } = render(
			<Provider store={store}>
				<I18nextProvider i18n={i18n}>
					<DraggableTableRow index={2} moveRow={jest.fn()} />
				</I18nextProvider>
			</Provider>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
