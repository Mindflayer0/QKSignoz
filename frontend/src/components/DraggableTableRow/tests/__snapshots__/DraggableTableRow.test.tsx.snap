// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableTableRow Snapshot test should render DraggableTableRow 1`] = `<DocumentFragment />`;
